File: app13_ruby_FastApi/backend/db/migrations/007_create_content_hashes.rb
# frozen_string_literal: true

Sequel.migration do
  change do
    create_table(:content_hashes) do
      String :file_name, primary_key: true
      String :sha256_hash, null: false
      Time :verified_at, null: false
      TrueClass :is_valid, null: false
      String :verified_by, null: false, default: "rubyguard-integrity-checker"
    end
  end
end

File: app14_LUA_UNITY/backend/db/migrations/007_create_content_hashes.lua
local schema = require("lapis.db.schema")
local types = schema.types

return {
  up = function()
    schema.create_table("content_hashes", {
      { "file_name", types.varchar({ primary_key = true }) },
      { "sha256_hash", types.varchar({ null = false }) },
      { "verified_at", types.time({ null = false }) },
      { "is_valid", types.boolean({ null = false }) },
      { "verified_by", types.varchar({ null = false, default = "luaguard-integrity-checker" }) }
    })
  end,
  down = function()
    schema.drop_table("content_hashes")
  end
}
